decorator.rb

```diff
@@ -0,0 +1,25 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    @nameable = nameable
+    super()
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.capitalize
+  end
+end
+
+class TrimmerDecorator < Decorator
+  def correct_name
+    name = @nameable.correct_name
+    name.length > 10 ? name[0..9] : name
+  end
+end
```

Comment:
- [ ] For your code to be modular and readable, I strongly recommend you separate these two classes into separate files. So create a capitalize_decorator.rb and trimmer_decorator.rb file.